### PR TITLE
1713/remove business logic from AppStatusItem component

### PR DESCRIPTION
**Pull request by emilyjablonski**
# Pull Request Template

## Issue

Addresses #1713

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Removes business logic from AppStatusItem component. Included removing our URL formatting. Also fixes a bug where the component would show `Application Due Date: Invalid Date` if the date was null (any FCFS listing) - now I just don't show that piece if it's null.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Prototype/POC (not to merge)
- [x] This change is a refactor/address technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

## How Can This Be Tested/Reviewed?

You can look in storybook, and in the public app the AppStatusItem component is used in the My Applications tab.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [x] I have updated the changelog to include a description of my changes

### CODE REVIEW

File: ui-components/src/blocks/AppStatusItem.tsx [hunk 1]
```diff
@@ -45,25 +44,22 @@ const AppStatusItem = (props: AppStatusItemProps) => {
...
           </div>
         </section>
 
         <footer className="status-item__footer">
           <div className="status-item_links">
-            <LocalizedLink
-              className="status-item__link lined"
-              href={`/listing/${listing.id}/${listing.urlSlug}`}
-            >
+            <LocalizedLink className="status-item__link lined" href={props.listingURL}>
```
Comment (jaredcwhite) [inline ui-components/src/blocks/AppStatusItem.tsx:55]: FYI, LocalizedLink is deprecated and we should probably remove it soon. Best to use `LinkComponent` from the `NavigationContext` instead. (See the `LinkButton` component for example usage.)

File: ui-components/src/blocks/AppStatusItem.tsx [hunk 2]
```diff
@@ -1,40 +1,39 @@
...
-  listing: Listing
+  applicationDueDate?: string
+  applicationURL: string
+  applicationUpdatedAt: string
+  confirmationNumber?: string
+  listingName: string
+  listingURL: string
 }
 
 const AppStatusItem = (props: AppStatusItemProps) => {
```
Comment (jaredcwhite) [inline ui-components/src/blocks/AppStatusItem.tsx:15]: I wonder if we should rename this and the props as well to something even more generic like just `StatusItem`. That's what the CSS class names are already.

**Review by jaredcwhite — CHANGES_REQUESTED** [review]
Just one deprecation to look at and a general question, otherwise looks good to me. :+1:

**Review by seanmalbert — APPROVED** [review]
Looks good to me, @jaredcwhite , can you take another look since the updates.

**Comment by emilyjablonski** [comment]
@jaredcwhite

**Review by jaredcwhite — APPROVED** [review]
LGTM :+1:

**Pull request merged**